projects/packages/forms/src/contact-form/class-contact-form-endpoint.php

```diff
@@ -986,4 +1001,44 @@ public function disable_integration( $request ) {
 		$is_deleted = External_Connections::delete_connection( $slug );
 		return rest_ensure_response( array( 'deleted' => $is_deleted ) );
 	}
+
+	/**
+	 * Return consolidated Forms config payload.
+	 *
+	 * @param WP_REST_Request $request Request.
+	 * @return WP_REST_Response
+	 */
+	public function get_forms_config( WP_REST_Request $request ) { // phpcs:ignore VariableAnalysis.CodeAnalysis.VariableAnalysis.UnusedVariable
+		$switch  = new Dashboard_View_Switch();
+		$feature = Jetpack_AI_Helper::get_ai_assistance_feature();
+		$has_ai  = ! is_wp_error( $feature ) ? ( $feature['has-feature'] ?? false ) : false;
+
+		$config = array(
+			// From jpFormsBlocks in class-contact-form-block.php.
+			'formsResponsesUrl'       => $switch->get_forms_admin_url(),
+			'preferredView'           => $switch->get_preferred_view(),
+			'isMailPoetEnabled'       => Jetpack_Forms::is_mailpoet_enabled(),
+			// From config in class-dashboard.php.
+			'blogId'                  => get_current_blog_id(),
+			'gdriveConnectSupportURL' => esc_url( Redirect::get_url( 'jetpack-support-contact-form-export' ) ),
+			'pluginAssetsURL'         => Jetpack_Forms::assets_url(),
+			'siteURL'                 => ( new Status() )->get_site_suffix(),
+			'hasFeedback'             => ( new Forms_Dashboard() )->has_feedback(),
+			'hasAI'                   => $has_ai,
+			'enableIntegrationsTab'   => apply_filters( 'jetpack_forms_enable_integrations_tab', true ),
+			'renderMigrationPage'     => $switch->is_jetpack_forms_announcing_new_menu(),
+			'dashboardURL'            => add_query_arg( 'jetpack_forms_migration_announcement_seen', 'yes', $switch->get_forms_admin_url() ),
+			// New data.
+			'canInstallPlugins'       => current_user_can( 'install_plugins' ),
+			'canActivatePlugins'      => current_user_can( 'activate_plugins' ),
+		);
+
+		// Include dashboard-only nonces when in admin Forms page context.
+		if ( is_admin() && isset( $_GET['page'] ) && 'jetpack-forms-admin' === $_GET['page'] ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
```

Comment:
I think we can always generate these even if they are only used in the dashboard.